### ISSUE TITLE
Class orders mismatch

### ISSUE THREAD

**Issue by illw84u** [experiencor/keras-yolo3#82]
it seems somebody else has already mentioned in previous thread.  When I trained on my own multi-class dataset, predict.py gave wrong labels on every class. I think it's because the order of classes in the Jason file is messed up. Any thoughts?

**Comment by jamartinh** [comment]
Try to delete the "*_train.pkl " that is generated every time you call train. If it exists, information on new labels and config is not read again.

**Comment by illw84u** [comment]
@jamartinh Deleting the pkl file didn't change the labels. I will try to do some research today. Let's see. Anybody else got this problem?

**Comment by illw84u** [comment]
solved the problem by adding 
labels = sorted(labels) 
somewhere in draw_boxes() in bbox.py

or delete sorted() when return for func create_training_instances()
`def create_training_instances(
    train_annot_folder,
    train_image_folder,
    train_cache,
    valid_annot_folder,
    valid_image_folder,
    valid_cache,
    labels,
):
    # parse annotations of the training set
    train_ints, train_labels = parse_voc_annotation(train_annot_folder, train_image_folder, train_cache, labels)

    # parse annotations of the validation set, if any, otherwise split the training set
    if os.path.exists(valid_annot_folder):
        valid_ints, valid_labels = parse_voc_annotation(valid_annot_folder, valid_image_folder, valid_cache, labels)
    else:
        print("valid_annot_folder not exists. Spliting the trainining set.")

        train_valid_split = int(0.8*len(train_ints))
        np.random.seed(0)
        np.random.shuffle(train_ints)
        np.random.seed()

        valid_ints = train_ints[train_valid_split:]
        train_ints = train_ints[:train_valid_split]

    # compare the seen labels with the given labels in config.json
    if len(labels) > 0:
        overlap_labels = set(labels).intersection(set(train_labels.keys()))

        print('Seen labels: \t'  + str(train_labels) + '\n')
        print('Given labels: \t' + str(labels))

        # return None, None, None if some given label is not in the dataset
        if len(overlap_labels) < len(labels):
            print('Some labels have no annotations! Please revise the list of labels in the config.json.')
            return None, None, None
    else:
        print('No labels are provided. Train on all seen labels.')
        print(train_labels)
        labels = train_labels.keys()

    max_box_per_image = max([len(inst['object']) for inst in (train_ints + valid_ints)])

    return train_ints, valid_ints, labels, max_box_per_image`
